ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandRunnerTest.java

```diff
@@ -167,6 +182,75 @@ public void shouldEarlyOutIfNewCommandsContainsTerminate() {
     verify(statementExecutor, never()).handleRestore(queuedCommand3);
   }
 
+  @Test
+  public void shouldReportRunningIfNotStuckProcessingCommand() throws BrokenBarrierException, InterruptedException, ExecutionException {
+    try {
+      checkCommandRunnerStatus(
+          COMMAND_RUNNER_HEALTH_TIMEOUT - 500,
+          COMMAND_RUNNER_HEALTH_TIMEOUT - 1000,
+          CommandRunner.CommandRunnerStatus.RUNNING
+      );
+    } catch (Exception e) {
+      // fail test if an exception happens
+      assertThat(true, equalTo(false));
+    }
+  }
+
+  @Test
+  public void shouldReportErrorIfStuckProcessingCommand() throws BrokenBarrierException, InterruptedException, ExecutionException {
+    try {
+      checkCommandRunnerStatus(
+          COMMAND_RUNNER_HEALTH_TIMEOUT + 1000,
+          COMMAND_RUNNER_HEALTH_TIMEOUT + 500,
+          CommandRunner.CommandRunnerStatus.ERROR
+      );
+    } catch (Exception e) {
+      // fail test if an exception happens
+      assertThat(true, equalTo(false));
+    }
+  }
+
+  private void checkCommandRunnerStatus(
+      long commandProcessingTimeMs,
+      long timeToCheckMetricMs,
+      CommandRunner.CommandRunnerStatus expectedStatus
+  ) throws BrokenBarrierException, InterruptedException, ExecutionException {
+    // Given:
+    givenQueuedCommands(queuedCommand1);
+    doAnswer((Answer) invocation -> {
+      Thread.sleep(commandProcessingTimeMs);
+      return null;
+    }).when(statementExecutor).handleStatement(queuedCommand1);
+
+    // When:
+    final CyclicBarrier gate = new CyclicBarrier(3);
```

Comment:
what's the point of this gate? to get all the threads to start doing their work around the same time?

Reply:
^ yea that was the idea behind it.

Reply:
the gate has been removed